src/registrar/tests/test_views.py

```diff
@@ -1222,6 +1233,55 @@ def test_domain_invitation_flow(self):
         home_page = self.app.get(reverse("home"))
         self.assertContains(home_page, self.domain.name)
 
+    def test_domain_nameservers(self):
+        """Can load domain's nameservers page."""
+        page = self.client.get(
+            reverse("domain-nameservers", kwargs={"pk": self.domain.id})
+        )
+        self.assertContains(page, "Domain name servers")
+
+    def test_domain_nameservers_form(self):
+        """Can change domain's nameservers.
+
+        Uses self.app WebTest because we need to interact with forms.
+        """
+        nameservers_page = self.app.get(
+            reverse("domain-nameservers", kwargs={"pk": self.domain.id})
+        )
+        session_id = self.app.cookies[settings.SESSION_COOKIE_NAME]
+        self.app.set_cookie(settings.SESSION_COOKIE_NAME, session_id)
+        with less_console_noise():  # swallow log warning message
+            result = nameservers_page.form.submit()
+        # form submission was a post, response should be a redirect
+        self.assertEqual(result.status_code, 302)
+        self.assertEqual(
+            result["Location"],
+            reverse("domain-nameservers", kwargs={"pk": self.domain.id}),
+        )
+        self.app.set_cookie(settings.SESSION_COOKIE_NAME, session_id)
+        page = result.follow()
+        self.assertContains(page, "The name servers for this domain have been updated")
+
+    def test_domain_nameservers_form_invalid(self):
+        """Can change domain's nameservers.
+
+        Uses self.app WebTest because we need to interact with forms.
+        """
+        nameservers_page = self.app.get(
+            reverse("domain-nameservers", kwargs={"pk": self.domain.id})
+        )
+        session_id = self.app.cookies[settings.SESSION_COOKIE_NAME]
+        self.app.set_cookie(settings.SESSION_COOKIE_NAME, session_id)
+        # first two nameservers are required, so if we empty one out we should
+        # get a form error
+        nameservers_page.form["form-0-server"] = ""
```

Comment:
We (@abroddrick @brandonlenz and me) are wondering are you getting "form-0-server" from?

Reply:
This is the ID of the first form input element on the page. Django-webtest (that's what `self.app` is) https://github.com/django-webtest/django-webtest addresses the form fields by their ids https://docs.pylonsproject.org/projects/webtest/en/latest/forms.html#filling-a-form. In this case, I think that I got that ID by inspecting the page in my browser. ;)